jvm/src/test/scala/cats/tests/TaskSpec.scala

```diff
@@ -0,0 +1,159 @@
+package cats
+package jvm
+package tests
+
+import cats.tests.CatsSuite
+import cats.data.{Streaming,Xor}
+
+class TaskTests extends CatsSuite {
+
+  test("Task should not run side-effects until .unsafePerformIO() is called") {
+    var run: Int = 0
+    val t: Task[Int] = Task.later { run += 1; run }
+    run should be (0)
+    t.unsafePerformIO()
+    run should be (1)
+  }
+
+  test("Task should only evaulate side-effects once for Task.later") {
+    var run: Int = 0
+    val t: Task[Int] = Task.later { run += 1; run }
+    t.unsafePerformIO()
+    run should be(1)
+
+    t.unsafePerformIO()
```

Comment:
As an alternative to simulating asynchronous execution and time (which is much cleaner), on top of Scala.js we can also have real asynchronous tests, which my own [Minitest](https://github.com/monixio/minitest) supports.